docs/dev/agent_dev_env.md

```diff
@@ -28,11 +33,44 @@ You will also need the Visual Studio for [Visual Studio for Python installer](ht
 
 Download the [gcc toolchain](http://win-builds.org/).
 - From the graphical package manager, select and install the needed libraries, leave the default (select all) if you're unsure.
-- Make sure to select x86_64.
-- Add installation folder to the %PATH%.
+- Make sure to select `x86_64`.
+- Add installation folder to the `%PATH%`.
+
+
+### Python Dependencies
+
+#### Preface
+
+To protect and isolate your system-wide python installation, a python virtual
+environment is _highly_ recommended (though optional). It will help keep a
+self-contained development environment and ensure a clean system Python.
+
+**Please note that due to the [way
+some virtual environments handle executable paths](https://bugs.python.org/issue22213)
+(e.g. `python -m venv`), not all vrtual environment options will be able to run the built
+agent correctly. At this time, the only confirmed virtual enviroment that is known for
+sure to work is `virtualenv`.**
+
+- Install the virtualenv module:
+`pip3 install virtualenv`
+- Create the virtual environment:
+`virtualenv $GOPATH/src/github.com/DataDog/datadog-agent/venv`
+- [Activate the virtualenv](https://virtualenv.pypa.io/en/latest/user_guide.html#activators) (OS-dependent).
+  This must be done for every new terminal before you start.
+- Specify the path when building the agent:
+`invoke agent.build --python-home-3=$GOPATH/src/github.com/DataDog/datadog-agent/venv`
```

Comment:
Is this needed ? It should be handle automatically when using a venv, no ?
On linux I don't need this.

If it's not automatic on OSX we should handle the case. The goal is to be able do build the agent with only `invoke agent.build`

Reply:
As far as I could figure it out on OSX, the packages are still attempted only to be loaded from system store even if you use a virtualenv. I'll update the docs to make sure to note that this only applies to that platform.